src/parser/mod.rs

```diff
@@ -2607,6 +2591,56 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_variant_object_key(&mut self) -> Result<VariantPathElem, ParserError> {
+        match self.next_token() {
+            TokenWithLocation {
+                token:
+                    Token::Word(Word {
+                        value,
+                        quote_style: quote_style @ (Some('"') | None),
+                        // Some experimentation suggests that snowflake permits
+                        // any keyword here unquoted.
+                        keyword: _,
+                    }),
+                ..
+            } => Ok(VariantPathElem::Dot {
+                key: value,
+                quoted: quote_style.is_some(),
+            }),
+            token => self.expected("variant object key name", token),
+        }
+    }
+
+    fn parse_variant_access(&mut self, expr: Expr) -> Result<Expr, ParserError> {
+        let mut path = Vec::new();
+        loop {
+            match self.next_token().token {
+                Token::Colon if path.is_empty() => {
+                    path.push(self.parse_variant_object_key()?);
+                }
```

Comment:
A variant access can begin with either bracket or "dot" syntax (the colon) corresponding to the two sites where this function is called, so the `:` token is not necessarily required initially. I can put the `:` handling  as a special case up front though to avoid the `is_empty` checks in the loop.

Reply:
Yeah I was thinking rather that the bracket syntax is identical to the existing map expr so it would be reasonable to call that function instead of this one (hence the peek_token beforehand)